packages/reporters/cli/src/CLIReporter.js

```diff
@@ -1,16 +1,142 @@
 // @flow strict-local
 
-import nullthrows from 'nullthrows';
+import type {
+  BundleGraph,
+  LogEvent,
+  ParcelOptions,
+  ProgressLogEvent,
+  ReporterEvent
+} from '@parcel/types';
+
+import {prettifyTime} from '@parcel/utils';
+import {render} from 'ink';
 import {Reporter} from '@parcel/plugin';
 import React from 'react';
-import {render} from 'ink';
+
+import {getProgressMessage} from './utils';
+import logLevels from './logLevels';
 import UI from './UI';
 
-let ui: ?UI;
-render(<UI ref={u => (ui = u)} />);
+type State = {|
+  progress: ?ProgressLogEvent,
+  logs: Array<LogEvent>,
+  bundleGraph: ?BundleGraph
+|};
+
+let state: State = {
```

Comment:
this exposes the true global nature of this state (it was before through the global `UI` object`).

Maybe we should allow plugins to export factory functions that return disposables instead? Or even just a class reference? cc @jamiebuilds @jamiekyle-eb on the RFC